### PR TITLE
[Helper] ADD option to cmake for advanced timer

### PR DESCRIPTION
**Pull request by bakpaul**
Add the `SOFA_SCOPED_ADVANCED_TIMER` cmake option to add the contruction of a `sofa::helper::ScopedAdvancedTimer` in the macro `SCOPED_TIMER` and `SCOPED_TIMER_VARNAME` along or in the place of the Tracy timer. 

This allows to also compile no timer at all.
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/framework/Config/CMakeLists.txt [hunk 1]
```diff
@@ -259,6 +259,12 @@ if (SOFA_TRACY)
     message(STATUS "SOFA is compiled with the Tracy profiler client. Use the Tracy server ${SOFA_TRACY_VERSION}.")
 endif()
 
+
+option(SOFA_SCOPED_ADVANCED_TIMER "Compile SOFA with the scoped advanced timer" ON)
```
Comment (fredroy) [inline Sofa/framework/Config/CMakeLists.txt]: Calling "scoped advanced timer" the "default timer" in SOFA looks weird to me, especially for the lambda user.
I would say
```suggestion
option(SOFA_ENABLE_BUILTIN_TIMER "Enable the builtin timers" ON)
```
for example, or something like that.

"Scoped" refer to a C++ thingy and Advanced... I dont know why 😅

File: Sofa/framework/Config/CMakeLists.txt [hunk 2]
```diff
@@ -259,6 +259,12 @@ if (SOFA_TRACY)
...
+if(SOFA_SCOPED_ADVANCED_TIMER)
+    set(ENABLE_SCOPED_ADVANCED_TIMER 1)
```
Comment (fredroy) [inline Sofa/framework/Config/CMakeLists.txt]:
```suggestion
    set(SOFA_ENABLE_SCOPED_ADVANCED_TIMER 1)
```
usually, "global" macro variable like that has `SOFA_` as a prefix

**Review by fredroy — CHANGES_REQUESTED** [review]

File: Sofa/framework/Config/CMakeLists.txt (hunk 1 above)
Comment (bakpaul) [reply Sofa/framework/Config/CMakeLists.txt]: Yes you are right, I'm not good with names...

**Review by alxbilger — APPROVED** [review]

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Review by epernod — APPROVED** [review]

**Pull request merged**